### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'group'

### ISSUE THREAD

**Issue by halleyshx** [ssut/py-googletrans#48]
My script like this:
#-*- coding:utf-8 -*-
from googletrans import Translator
import sys
import io

translator = Translator()
tanstext = u'我们'
tran_ch = translator.translate(tanstext ).text
print tran_ch
###############
yesterday this is ok and no problem，but today I run my script without internet, show some problem.And I run the script in the internet environment, but it's show same problem,can't run. 
The cmd show the Traceback like this:
Traceback (most recent call last):
  File "D:\JDstory\workstory\translation-project\filetocut\trans-test.py", line
10, in <module>
    tran_ch = translator.translate(tanstext).text
  File "D:\software\Python27\lib\site-packages\googletrans\client.py", line 132,
 in translate
    data = self._translate(text, dest, src)
  File "D:\software\Python27\lib\site-packages\googletrans\client.py", line 57,
in _translate
    token = self.token_acquirer.do(text)
  File "D:\software\Python27\lib\site-packages\googletrans\gtoken.py", line 180,
 in do
    self._update()
  File "D:\software\Python27\lib\site-packages\googletrans\gtoken.py", line 59,
in _update
    code = unicode(self.RE_TKK.search(r.text).group(1)).replace('var ', '')
AttributeError: 'NoneType' object has no attribute 'group'
#########################
That's why?  Thank you so much!

**Comment by Andy671** [comment]
Have you found a solution?

**Comment by Larkina** [comment]
This means google banned your IP.

**Comment by Darkblader24** [comment]
Wrong, this happens when Google directly sends you the raw token. They do this for a short time after maintenance, not sure why.

**Comment by MTL1991** [comment]
Hi,

from yesterday I have this "error". How do you think that I could fix it. I just created other server to execute my code and the result it's the same :(

Thanks

**Comment by Darkblader24** [comment]
See here for the fix: #78

**Comment by ariefsibuea** [comment]
how to use it in translation code above?

**Comment by romanseer** [comment]
go to browser with Google Translate page, try to translate Hello World =) 
if you see Error translated - your IP are banned

**Comment by ariefsibuea** [comment]
I have tried to translate from Google Translate page and it works. But when I try code above it failed with same error as shown above.

**Comment by ariefsibuea** [comment]
> See here for the fix: #78

Can I use the code with the translation code by @halleyshx ?

**Comment by hrishikeshpatel** [comment]
After applying #78 I got below error. Please help

/home/hrishikesh/anaconda3/envs/MTP_IITM/lib/python2.7/json/decoder.pyc in raw_decode(self, s, idx)
    380             obj, end = self.scan_once(s, idx)
    381         except StopIteration:
--> 382             raise ValueError("No JSON object could be decoded")
    383         return obj, end

ValueError: No JSON object could be decoded

**Comment by joaonetto** [comment]
Looking other places, probably the issue is around a Regex from Google Translate than they use to working on.

Take a look in: https://github.com/pndurette/gTTS/issues/138

**Comment by Maldus512** [comment]
> See here for the fix: #78

Apologies, I just installed this library via pip and encountered this error. How should I proceed to apply said fix?

**Comment by Creator-360** [comment]
hello, I try today to use the gtts on python but is not working
can somebody help me please ?

the error is:

Exception has occurred: OSError
[Errno 22] Invalid argument: 'C:\test.mp3'
File "C:\Users\Administrator\Desktop\Speech-Recognition.py", line 39, in
speech.save("C:\test.mp3")

My code:

import speech_recognition as sr
from gtts import gTTS
import os
import playsound

while True:

command = 0
speech = 0

mic = sr.Microphone()
r = sr.Recognizer()
with mic as source:
    print("Calibrate Microphone")
    print("Please wait...")
    r.adjust_for_ambient_noise(source, duration=1)
    print("listening...")
    audio = r.listen(source)

try:
    command = (r.recognize_google(audio))
    print("you said:"+command)

except sr.UnknownValueError:
    print("")


except sr.RequestError:
    print("")

if "hello" in command:
    print("hello")
    speech = gTTS("hello", lang="en")
    if os.path.exists("C:\test.mp3"):
        print("file exist")
        playsound.playsound("C:\test.mp3", True)
    else:
        print("saving new mp3 sound")
        speech.save("C:\test.mp3")
        playsound.playsound("C:\test.mp3", True)

**Comment by algoscale1** [comment]
@Darkblader24 My IP is also getting blocked after using the pull request [(https://github.com/ssut/py-googletrans/pull/78)], although it resolved the "None Type" issue. It happens after 250 requests or something, I have even tried on a different server, the IP is continuously getting blocked after few requests. Earlier I could make thousands of requests and my IP never got blocked. The new pull request has messed up something, any idea?

**Comment by ramya-ramanathan** [comment]
Quick question: What is the error you get when your IP is blocked? 

Is it `ValueError("No JSON object could be decoded")` ?

**Comment by Colanim** [comment]
The error when the IP is blocked is `Expecting value: line 1 column 1 (char 0)`

Any idea how to unblock my IP address ?

**Comment by 1010Manoj** [comment]
@ramya-ramanathan  I am getting ValueError: No JSON object could be decoded
what is this for???

**Comment by Liguo-Ma** [comment]
Hello
I get 'NoneType' object has no attribute 'group'
I have tried this
$ pip uninstall googletrans
$ git clone https://github.com/BoseCorp/py-googletrans.git
$ cd ./py-googletrans
$ python setup.py install
But get same issue
Please help me

**Comment by bastienboutonnet** [comment]
I've been getting this as well on a recent install of `googletrans` in a fresh venv:
```python
  File "<truncated_for_privacy>/lib/python3.6/site-packages/googletrans/client.py", line 182, in translate
    data = self._translate(text, dest, src, kwargs)
  File "<truncated_for_privacy>/lib/python3.6/site-packages/googletrans/client.py", line 78, in _translate
    token = self.token_acquirer.do(text)
  File "<truncated_for_privacy>/lib/python3.6/site-packages/googletrans/gtoken.py", line 194, in do
    self._update()
  File "<truncated_for_privacy>/lib/python3.6/site-packages/googletrans/gtoken.py", line 62, in _update
    code = self.RE_TKK.search(r.text).group(1).replace('var ', '')
AttributeError: 'NoneType' object has no attribute 'group'
```

**Comment by yoyoidea** [comment]
> Hello
> I get 'NoneType' object has no attribute 'group'
> I have tried this
> $ pip uninstall googletrans
> $ git clone https://github.com/BoseCorp/py-googletrans.git
> $ cd ./py-googletrans
> $ python setup.py install
> But get same issue
> Please help me

me, too.

**Comment by Hamilkar247** [comment]
still problem exist :'(

**Comment by waelmas** [comment]
Still having this issue too frequently. Impacting our production.

For days we thought we've been doing something wrong...

**Comment by ashishalakhani** [comment]
I am seeing this issue as well.  I see it on dev and staging environment but not in production.

**Comment by markklick206** [comment]
Getting the same "Attribute Error" after GoogleTrans had been working great for months...

```
  File "C:\Anaconda3\lib\site-packages\googletrans\gtoken.py", line 65, in _update
    code = unicode(self.RE_TKK.search(r.text).group(1)).replace('var ', '')
AttributeError: 'NoneType' object has no attribute 'group'
```

**Comment by mateoToquica** [comment]
I've been using this library to do research work, where I have to translate several images, at first I thought it was an error in the images because they have different edges and colors, but then I used a time delay for some images that did not work and suddenly they started translating. The problem is that it does not work for all the images, below I show how I use the delay

`
def translateImg(src):
    img = cv2.imread(src)
    text = pytesseract.image_to_string(img, lang='ara')
    print(text)
    time.sleep(10) # sleep the aplication 10 seconds
    translator = Translator()
    result = translator.translate(text)
    print(result.text)
for x in range(1,28):
    translateImg('../resources/photos/annunciAmaq1/img ('+str(x)+').jpg')
`

**Comment by jmagdeska** [comment]
I'm also getting the same issue, any fix for this?

```
File "/home/anaconda3/lib/python3.7/site-packages/googletrans/gtoken.py", line 62, in _update
    code = self.RE_TKK.search(r.text).group(1).replace('var ', '')
AttributeError: 'NoneType' object has no attribute 'group'
```

I'm using Python 3.7.3 on Ubuntu 20.04 machine

**Comment by MartinVinter** [comment]
Hey there,

In another 2 years old thread talking about this issue (which sadly, I cannot find again) it was mentioned that upgrading GoogleTrans helps. It has helped for me too, but only for a short while. In other old threads I have found, it was mentioned, that an older version of the issue was related to a problem with an API key, token or similar. So my theory is that I get another key if I update. Even when there isn't a new version, Anaconda will re-download and install the package. At least you can give it a shot. It is quite easy - run this in Anaconda:

`pip install git+https://github.com/BoseCorp/py-googletrans.git --upgrade`

**Comment by sergiomora03** [comment]
Hi, I get the same error.

Python version 3.8.3

**Comment by sergiomora03** [comment]
> Hey there,
> 
> In another 2 years old thread talking about this issue (which sadly, I cannot find again) it was mentioned that upgrading GoogleTrans helps. It has helped for me too, but only for a short while. In other old threads I have found, it was mentioned, that an older version of the issue was related to a problem with an API key, token or similar. So my theory is that I get another key if I update. Even when there isn't a new version, Anaconda will re-download and install the package. At least you can give it a shot. It is quite easy - run this in Anaconda:
> 
> `pip install git+https://github.com/BoseCorp/py-googletrans.git --upgrade`

It´s not work. Still the error:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-f71bcb40be3a> in <module>
----> 1 translator.translate('Hola Mundo')

~\Anaconda3\lib\site-packages\googletrans\client.py in translate(self, text, dest, src)
    170 
    171         origin = text
--> 172         data = self._translate(text, dest, src)
    173 
    174         # this code will be updated when the format is changed.

~\Anaconda3\lib\site-packages\googletrans\client.py in _translate(self, text, dest, src)
     73             text = text.decode('utf-8')
     74 
---> 75         token = self.token_acquirer.do(text)
     76         params = utils.build_params(query=text, src=src, dest=dest,
     77                                     token=token)

~\Anaconda3\lib\site-packages\googletrans\gtoken.py in do(self, text)
    184 
    185     def do(self, text):
--> 186         self._update()
    187         tk = self.acquire(text)
    188         return tk

~\Anaconda3\lib\site-packages\googletrans\gtoken.py in _update(self)
     63 
     64         # this will be the same as python code after stripping out a reserved word 'var'
---> 65         code = unicode(self.RE_TKK.search(r.text).group(1)).replace('var ', '')
     66         # unescape special ascii characters such like a \x3d(=)
     67         if PY3:  # pragma: no cover

AttributeError: 'NoneType' object has no attribute 'group'
```

**Comment by BoyuChen118** [comment]
You can simply overcome this issue by using error handling like so:
```
while True:
    try:
       # you translation code here
    except:
        print(Exception)
        continue
    break
```

**Comment by ashishalakhani** [comment]
> You can simply overcome this issue by using error handling like so:
> 
> ```
> while True:
>     try:
>        # you translation code here
>     except:
>         print(Exception)
>         continue
>     break
> ```

Won't this go in infinite loop?

**Comment by NicoCaldo** [comment]
> > You can simply overcome this issue by using error handling like so:
> > ```
> > while True:
> >     try:
> >        # you translation code here
> >     except:
> >         print(Exception)
> >         continue
> >     break
> > ```
> 
> Won't this go in infinite loop?

For some reason, at a certain point, the translation succeed, at least, from my experience

**Comment by jmagdeska** [comment]
> > You can simply overcome this issue by using error handling like so:
> > ```
> > while True:
> >     try:
> >        # you translation code here
> >     except:
> >         print(Exception)
> >         continue
> >     break
> > ```
> 
> Won't this go in infinite loop?

Yes, issue is not solved, the program goes to infinite loop.

**Comment by ashishalakhani** [comment]
I don't think we can rely on this library for production use. Any other free alternative?

**Comment by NicoCaldo** [comment]
> I don't think we can rely on this library for production use. Any other free alternative?

It works like a charm for me https://github.com/lushan88a/google_trans_new

**Comment by jmagdeska** [comment]
> > I don't think we can rely on this library for production use. Any other free alternative?
> 
> It works like a charm for me https://github.com/lushan88a/google_trans_new

Thank you for sharing this resource, I confirm it works without problems so far.

**Comment by ohMuzzy** [comment]
> 
> 
> > I don't think we can rely on this library for production use. Any other free alternative?
> 
> It works like a charm for me https://github.com/lushan88a/google_trans_new

Be careful this is not Google Translate the output is much lower quality to what you get at https://translate.google.com/

There is a working solution for AHK (https://www.autohotkey.com/boards/viewtopic.php?t=63835). Why can't we make it work in Python?

**Comment by subhrasmitdas123** [comment]
Hello
I  tried to make language translator app using the latest version and it worked but whenever i tried to run it any system it is showing a none type error and to resolve it i unistall the version and tried to run it 3.1.0a0 version and same it is working on my side but the error is the same while running it in other system . what is the solution to it ?

**Comment by apollo1234567890** [comment]
Hi !! the same problem. 
 line 62, in _update
     code = self.RE_TKK.search(r.text).group(1).replace('var ', '')
 AttributeError: 'NoneType' object has no attribute 'group'
